godtools/App/Share/Data/FavoritedResourcesRepository/Cache/RealmFavoritedResourcesCache.swift

```diff
@@ -191,43 +184,48 @@ class RealmFavoritedResourcesCache {
     
     // MARK: - Private
     
-    private func getFavoritesSortedByPosition() -> [RealmFavoritedResource] {
+    private func getFavoritesSortedByPositionPublisher() -> AnyPublisher<[RealmFavoritedResource], Never> {
+        
         let realm = realmDatabase.openRealm()
+        
         let favoritesSortedByPosition = Array(
             realm.objects(RealmFavoritedResource.self)
             .sorted(byKeyPath: #keyPath(RealmFavoritedResource.position), ascending: true)
-            )
+        )
         
         if favoritesSortedByPosition.allSatisfy({ $0.position == 0 }) {
             
-            return migrateExistingFavoritesToPositions()
+            return migrateExistingFavoritesToPositionsPublisher()
+                .replaceError(with: [])
+                .eraseToAnyPublisher()
             
         } else {
-            return favoritesSortedByPosition
+            return Just(favoritesSortedByPosition)
+                .eraseToAnyPublisher()
         }
     }
     
     @available(*, deprecated)
-    private func migrateExistingFavoritesToPositions() -> [RealmFavoritedResource] {
-        let realm = realmDatabase.openRealm()
-        let favoritesSortedByCreatedAt = realm
-            .objects(RealmFavoritedResource.self)
-            .sorted(byKeyPath: #keyPath(RealmFavoritedResource.createdAt), ascending: false)
-                    
-        do {
-            try realm.write {
```

Comment:
Hey @rachaelblue I'm about 95% certain this write crash was caused because the realm onChange is also triggered on write transactions. So Realm complained that there was a write within a write.

I've seen this happen before elsewhere.  That's one of the things that prompted me to create the RealmDatabase extensions and the ```writePublisher``` because that publisher will always create a new realm instance on another thread so it's confined within it's own thread.